### PR TITLE
[Controller] Emergency source region config scoped per cluster

### PR DESCRIPTION
**Pull request by asaxena76**
## Summary, imperative, start upper case, don't end with a period

[Controller] Emergency source region config should be scoped per cluster to help with testing and can also be useful with complex deployments

## How was this PR tested?

Existing integration tests
## Does this PR introduce any user-facing changes?

- [x ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 1]
```diff
@@ -7254,13 +7254,14 @@ public Optional<PushStatusStoreRecordDeleter> getPushStatusStoreRecordDeleter()
...
    */
   @Override
-  public Optional<String> getEmergencySourceRegion() {
-    return multiClusterConfigs.getEmergencySourceRegion().equals("")
+  public Optional<String> getEmergencySourceRegion(String clusterName) {
+
+    return multiClusterConfigs.getEmergencySourceRegion(clusterName).equals("")
         ? Optional.empty()
-        : Optional.of(multiClusterConfigs.getEmergencySourceRegion());
+        : Optional.of(multiClusterConfigs.getEmergencySourceRegion(clusterName));
```
Comment (gaojieliu) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:7264]: Nit:
Can we call this function: `multiClusterConfigs.getEmergencySourceRegion(clusterName)` just once?

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 2]
```diff
@@ -7254,13 +7254,14 @@ public Optional<PushStatusStoreRecordDeleter> getPushStatusStoreRecordDeleter()
...
   }
```
[the remark this answers is not in the document]
Comment (asaxena76) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:7265]: Yeh the optional a bit verbose but i wanted to keep it consistent with rest of the class. good point on stringutils.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 3]
```diff
@@ -7254,13 +7254,16 @@ public Optional<PushStatusStoreRecordDeleter> getPushStatusStoreRecordDeleter()
...
+   * @see Admin#getEmergencySourceRegion(String)
    */
   @Override
-  public Optional<String> getEmergencySourceRegion() {
-    return multiClusterConfigs.getEmergencySourceRegion().equals("")
-        ? Optional.empty()
-        : Optional.of(multiClusterConfigs.getEmergencySourceRegion());
+  public Optional<String> getEmergencySourceRegion(@Nonnull String clusterName) {
+    String emergencySourceRegion = multiClusterConfigs.getEmergencySourceRegion(clusterName);
+    if (StringUtils.isNotEmpty(emergencySourceRegion)) {
```
Comment (adamxchen) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:7262]: nit: can use one line for this
`return StringUtils.isNotEmpty(emergencySourceRegion) ? Optional.of(emergencySourceRegion): Optional.empty()` or even
`Optional.ofNullable(emergencySourceRegion).filter(s -> StringUtils.isNotEmpty(s));`

**Review by adamxchen — APPROVED** [review]
Looks good. Have one nit but it's okay to drop it.

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java (hunk 3 above)
Comment (asaxena76) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:7262]: Need to merge . Dropping this for now. Will have more opportunity to improve later

**Pull request merged**